autogen/main.tf

```diff
@@ -38,6 +38,7 @@ resource "random_shuffle" "available_zones" {
 locals {
   // location
   location = var.regional ? var.region : var.zones[0]
+  region   = var.region != null ? var.region : join("-", slice(split("-", var.zones[0]), 0, 2))
```

Comment:
This would be more clear if inverted -- e.g. `region = var.region == null ? join("-", slice(split("-", var.zones[0]), 0, 2)) : var.region`. As written, it's hard to tell exactly what's going on.

Reply:
Reworked